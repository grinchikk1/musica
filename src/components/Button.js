import React from "react";
import PropTypes from "prop-types";
import "../styles/Button.scss";

const Button = ({ backgroundColor, text, onClick }) => {
  return (
    <button className="button" style={{ backgroundColor }} onClick={onClick}>
      {text}
    </button>
  );
};

Button.propTypes = {
  backgroundColor: PropTypes.string,
  text: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default Button;
