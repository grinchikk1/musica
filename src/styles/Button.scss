.button {
  color: #fff;
  padding: 10px 20px;
  border-radius: 5px;
  border: none;
  cursor: pointer;
  margin: 10px;
  font-size: 16px;
  transition: background-color 0.3s ease-in-out;

  &:hover {
    opacity: 0.7;
  }
}
