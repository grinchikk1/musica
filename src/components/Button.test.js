import React from "react";
import renderer from "react-test-renderer";
import { screen, render, fireEvent } from "@testing-library/react";
import Button from "./Button";

describe("Button Component", () => {
  // snapshot test
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Button backgroundColor="goldenrod" text="Buy Now" onClick={() => {}} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  // unit test
  it("calls onClick handler when clicked", () => {
    const onClickMock = jest.fn();
    render(
      <Button text="Click me" onClick={onClickMock} />
    );
    fireEvent.click(screen.getByText("Click me"));
    expect(onClickMock).toHaveBeenCalled();
  });
});
