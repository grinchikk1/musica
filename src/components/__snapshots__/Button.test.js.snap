// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders correctly 1`] = `
<button
  className="button"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "goldenrod",
    }
  }
>
  Buy Now
</button>
`;
